internal/filesystem.go

```diff
@@ -508,3 +517,38 @@ func prefixPath(p string, envKey string) string {
 	}
 	return filepath.Clean(dir + p)
 }
+
+// Will open or create the given file.
+// If a file already exists with the given name but is not a regular file,
+// e.g. a symlink, it will be deleted and a regular file re-created instead
+func OpenOrCreateRegularFile(fileName string, flags int, permission fs.FileMode) (*os.File, error) {
+	fileName = filepath.Clean(fileName)
+	openFile := func() (*os.File, error) {
+		// check if it is a file before opening, because if it is a pipe will block on open
+		if !IsFile(fileName) {
+			if err := os.Remove(fileName); err != nil && !os.IsNotExist(err) {
+				return nil, fmt.Errorf("cannot delete file %s: %w", fileName, err)
+			}
+		}
+		// #nosec G304 -- fileName was cleaned before
+		file, err := os.OpenFile(fileName, flags, permission)
+		if err != nil {
+			return nil, fmt.Errorf("opening file %s: %w", fileName, err)
+		}
+
+		// check again that the file is a regular file after it is open to be sure it was not changed between checking and opening it
+		if !IsFile(fileName) {
+			return nil, fmt.Errorf("not a regular file %s", fileName)
+		}
+
+		return file, nil
+	}
+
+	file, err := openFile()
+	if err != nil {
+		// try one more time to open the file if it failed for the first time
+		return openFile()
```

Comment:
Why need to try 2 times? Any public recommendations or experimental findings? Why not 3 times?

Reply:
I've changed and now it is executed only once.
Initial idea was to try second time, in case file was deleted after it was opened.